### PR TITLE
Fix math notation for linear projections in nn.RNN docs

### PR DESCRIPTION
**Pull request by yashika-git**
Updating line number 316.

Fixes #76658

### CODE REVIEW

File: torch/nn/modules/rnn.py [hunk 1]
```diff
@@ -313,7 +313,7 @@ class RNN(RNNBase):
     function:
 
     .. math::
-        h_t = \tanh(W_{ih} x_t + b_{ih} + W_{hh} h_{(t-1)} + b_{hh})
+        h_t = \tanh(x_t * W_{ih}.T + b_{ih} + h_{(t-1)} * W_{hh}.T + b_{hh})
```
Comment (jbschlosser) [inline torch/nn/modules/rnn.py:316]: nit: to match the [`nn.Linear` docs](https://pytorch.org/docs/stable/generated/torch.nn.Linear.html), please leave out the `*` and use `^T` for the transpose notation.

**Review by jbschlosser — COMMENTED** [review]
Thanks for the contribution! Just a couple minor formatting tweaks to match the linear docs, then this should be good to go :)

File: torch/nn/modules/rnn.py (hunk 1 above)
Comment (yashika-git) [reply torch/nn/modules/rnn.py]: @jbschlosser I have done the required changes.

File: torch/nn/modules/rnn.py [hunk 2]
```diff
@@ -313,7 +313,7 @@ class RNN(RNNBase):
     function:
 
     .. math::
-        h_t = \tanh(W_{ih} x_t + b_{ih} + W_{hh} h_{(t-1)} + b_{hh})
+        h_t = \tanh((x_t)W_{ih}^T + b_{ih} + h_{(t-1)}W_{hh}^T + b_{hh})
```
Comment (jbschlosser) [inline torch/nn/modules/rnn.py:316]: Might be too pedantic, but I think it reads slightly better without the extra parens:

```suggestion
        h_t = \tanh(x_t W_{ih}^T + b_{ih} + h_{t-1}W_{hh}^T + b_{hh})
```

File: torch/nn/modules/rnn.py (hunk 1 above)
Comment (jbschlosser) [reply torch/nn/modules/rnn.py]: @yashika-git No failures that I've seen! We should wait for all jobs to complete before merging, however.

**Comment by yashika-git** [comment]
@jbschlosser Thank you for guiding throughout :)

**Review by jbschlosser — APPROVED** [review]

**Pull request closed without merging**